src/sentry/search/snuba/executors.py

```diff
@@ -333,6 +333,13 @@ def query(
             span.set_data("Result Size", len(group_ids))
         metrics.timing("snuba.search.num_candidates", len(group_ids))
 
+        project_orgs = set(project.organization for project in projects)
+        query_org = tuple(project_orgs)[0] if len(project_orgs) == 1 else None
```

Comment:
The organization isn't otherwise accessible here. It's used to define the `group_queryset` but can't be extracted without hacking apart the query itself.

I considered several kludges and am leaning toward this one. It's too bad that we'd get instrumentation only on queries that filter by at least one project (especially since a query for all projects would be bigger and therefore more informative) but I think it's good enough for the immediate goals.

The best other option I can think of would be to add an optional argument to inject the organization to this function, and then go pass it from the API endpoints that we're interested in. It would be an ugly bit of technical debt but we could clean it up after we're done with this feature flag.

There is no risk that a user would get the wrong flag, right? I believe the webapp should have enforced at a higher layer that the projects belong to the user's organization.